plaso/parsers/onedrive.py

```diff
@@ -0,0 +1,467 @@
+    decoded_parameters (str): decoded (and decrypted) parameters
+    raw_parameters (str): the raw parameters encoded as a hex string
+    recorded_time (dfdatetime.DateTimeValues): date and time the entry was
+        recorded.
+  """
+
+  DATA_TYPE = 'windows:onedrive:log'
+
+  def __init__(self):
+    """Initializes event data."""
+    super(OneDriveLogEvent, self).__init__(data_type=self.DATA_TYPE)
+    self.code_file_name = None
+    self.code_function_name = None
+    self.decoded_parameters = None
+    self.raw_parameters = None
+    self.recorded_time = None
+
+
+class OneDriveLogFileParser(
+    interface.FileEntryParser, dtfabric_helper.DtFabricHelper):
+  """Parser for OneDrive log files."""
+
+  NAME = 'onedrive_log'
+  DATA_FORMAT = 'OneDrive Log file'
+
+  BLOCK_SIGNATURE = b'\xcc\xdd\xee\xff\x00\x00\x00\x00'
+  COMPRESSED_BLOCK_SIGNATURE = b'\x1f\x8b\x08\x00\x00\x00\x00\x00'
+  FORMAT_SIGNATURE = b'EBFGONED'
+
+  _AES_IV = b'\x00' * 16
+  _BASE64_ALPHABET = (
+      'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789_-')
+  _DEFINITION_FILE = os.path.join(
+      os.path.dirname(__file__), 'onedrive.yaml')
+  _DELIMITERS = [r'%', r'\\', r'/', r'://', r'\?', r':', r'\.']
+  _DELIMITERS_REGEX_PATTERN = '({0:s})'.format('|'.join(_DELIMITERS))
+  _DELIMITERS_REGEX = re.compile(_DELIMITERS_REGEX_PATTERN)
+  _GENERAL_KEYSTORE_FILE = 'general.keystore'
+  _OBFUSCATED_STRING_MAP_FILE = 'ObfuscationStringMap.txt'
+
+  @classmethod
+  def GetFormatSpecification(cls):
+    """Retrieves the format specification.
+
+    Returns:
+      FormatSpecification: format specification.
+    """
+    format_specification = specification.FormatSpecification(cls.NAME)
+    format_specification.AddNewSignature(cls.FORMAT_SIGNATURE, offset=0)
+    return format_specification
+
+  def _GetFileEntryInSameDirectory(
+      self, sibling_entry, target_filename, file_system, parser_mediator):
+    """Gets a file entry from the same directory as another file.
+
+    Args:
+      target_filename (str): the name of the file to open.
+      sibling_entry (dfvfs.FileEntry): the sibling file entry.
+      file_system (dfvfs.FileSystem): the file system containing the OneDrive
+        logs.
+      parser_mediator (ParserMediator): mediates interactions between parsers
+        and other components, such as storage and dfVFS.
+
+    Returns:
+      dfvfs.FileEntry: the target file entry.  None when the target file is not
+          found or able to be opened.
+    """
+    path_segments = file_system.SplitPath(sibling_entry.path_spec.location)
+    path_segments.pop()
+    path_segments.append(target_filename)
+
+    location = file_system.JoinPath(path_segments)
+    kwargs = {
+        'location': location,
+        'parent': None}
+
+    filename_path_spec = path_spec_factory.Factory.NewPathSpec(
+        sibling_entry.path_spec.TYPE_INDICATOR, **kwargs)
+
+    try:
+      filename_file_entry = path_spec_resolver.Resolver.OpenFileEntry(
```

Comment:
this can be done differently. (1) get the parent file entry (2) use parent.GetSubFileEntryByName()

Reply:
Done